src/mono/mono/metadata/assembly.c

```diff
@@ -140,6 +140,18 @@ mono_set_assemblies_path (const char* path)
 	assemblies_path = dest = split;
 	while (*split) {
 		char *tmp = *split;
+		int len = strlen(tmp);
+		// If the current string ends with a backslash, append the next string to it
+		while (len > 0 && tmp[len - 1] == '\\') {
```

Comment:
I don't think it's necessarily a problem but the escaping mechanism seems rather inconsistent. This detects only backslash at the end of path that was followed by `:`. In Windows `\:` is never valid. In Linux it could be but then you can write it as `\\:` I guess.

Reply:
Ah, right, `G_SEARCHPATH_SEPARATOR_S` is going to be `;` on Windows. That makes it a bit more messy.

Reply:
Yes, the idea is to escape `\\:` on non-windows platforms which should allow correct path resolution. I parametrized the concat token and added `#ifndef HOST_WIN32` macro. Do you see any other issues with the escaping mechanism?